src/calendar-app/calendar/search/view/CalendarSearchView.ts

```diff
@@ -360,14 +351,39 @@ export class CalendarSearchView extends BaseTopLevelView implements TopLevelView
 		} else if (client.isCalendarApp()) {
 			return m.fragment({}, [
 				this.renderSearchResultActions(),
-				m(NavButton, {
-					label: "calendar_label",
-					hideLabel: true,
-					icon: () => BootIcons.Calendar,
-					href: CALENDAR_PREFIX,
-					centred: true,
-					fillSpaceAround: false,
+				m(IconButton, {
+					icon: Icons.Filter,
+					title: "filter_label",
+					click: () => {
+						const dialog = Dialog.editSmallDialog(
+							{
+								middle: () => lang.get("filter_label"),
+								right: [
+									{
+										label: "save_action",
+										click: () => {
+											dialog.close()
+										},
+										type: ButtonType.Primary,
+									},
+								],
+							},
+							() => m(".pt-xl.pb-xl", this.renderCalendarFilterSection()),
+						)
+						dialog.show()
+					},
 				}),
+				m(
```

Comment:
This is the button that was previously used to return to the calendar, right? I think we can remove it since it now uses the back button for this.